src/bundle.c

```diff
@@ -1898,7 +1898,7 @@
 
 	g_return_val_if_fail(bundlename, FALSE);
 	g_return_val_if_fail(bundle != NULL && *bundle == NULL, FALSE);
-	g_return_val_if_fail(!(params & TRUE), FALSE); /* protect against passing TRUE as the params enum */
+	g_return_val_if_fail(~(params & TRUE), FALSE); /* protect against passing TRUE as the params enum */
```

Comment:
Mh, the result of `param & TRUE` is interpreted here as a boolean anyway by `g_return_val_if_fail()`. We just invert it before.

Does this fix an issue for you, was that noted by any code scanning tool, or how did you come across this? This pattern is used earlier in the same method, too.

I am also unsure if code scanning complains about the original or the changed code here because the error message does not seem to match the changed line.

Reply:
@b4yuan The question of which concrete issue this fixes is still open.